pkg/build/api/validation/validation.go

```diff
@@ -64,14 +64,15 @@ func ValidateBuildConfig(config *buildapi.BuildConfig) field.ErrorList {
 	fromRefs := map[string]struct{}{}
 	specPath := field.NewPath("spec")
 	triggersPath := specPath.Child("triggers")
+	buildFrom := buildutil.GetImageStreamForStrategy(config.Spec.Strategy)
```

Comment:
Reminds me that we should use a subresource for that. We don't need the controller to be able to update the entire build config.